### ISSUE TITLE
updatedep failed due to checksum mismatch

### ISSUE THREAD

**Issue by WIZARD-CXY** [etcd-io/etcd#10437]
enviroment: 
go 1.11.2 on Mac 
code: master branch

use 
`$ ./scripts/updatedep.sh`
returns

```
go: verifying github.com/grpc-ecosystem/go-grpc-middleware@v1.0.0: checksum mismatch
        downloaded: h1:BWIsLfhgKhV5g/oF34aRjniBHLTZe5DNekSjbAjIS6c=
        go.sum:     h1:Iju5GlWwrvL6UBg4zJJt3btmonfrMlCDdsejg4CZE7c=
```

**Comment by hexfusion** [comment]
Hi @WIZARD-CXY thanks for the report because go-grpc-middleware does not support go mod every commit they make to master will change the checksum. On our end we can keep it updated but it seems this could be a regular thing and they really should resolve.

ref:
*  https://github.com/grpc-ecosystem/go-grpc-middleware/issues/185
* https://github.com/grpc-ecosystem/go-grpc-middleware/commits/master

**Comment by WIZARD-CXY** [comment]
I can't agree more.

**Comment by WIZARD-CXY** [comment]
It is kinda of pain in the ass, this mismatch will make the go mod command fail to run like `GO111MODULE=on go mod vendor -v` in updatedep.sh...

**Comment by hexfusion** [comment]
closed via #10439
